modules/monitoring/victoria-metrics/main.tf

```diff
@@ -0,0 +1,117 @@
+data "aws_eks_cluster" "this" {
+  name = var.cluster_name
+}
+
+data "aws_region" "current" {}
+
+resource "kubernetes_namespace" "this" {
+  depends_on = [
+    var.module_depends_on
+  ]
+  count = var.namespace == "" ? 1 - local.argocd_enabled : 0
+  metadata {
+    name = var.namespace_name
+  }
+}
+
+resource "local_file" "namespace" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content = yamlencode({
+    "apiVersion" = "v1"
+    "kind"       = "Namespace"
+    "metadata" = {
+      "name" = local.namespace
+    }
+  })
+  filename = "${path.root}/${var.argocd.path}/ns-${local.namespace}.yaml"
+}
+
+locals {
+  argocd_enabled = length(var.argocd) > 0 ? 1 : 0
+  namespace      = coalescelist(var.namespace == "" && local.argocd_enabled > 0 ? [{ "metadata" = [{ "name" = var.namespace_name }] }] : kubernetes_namespace.this, [{ "metadata" = [{ "name" = var.namespace }] }])[0].metadata[0].name
+}
+
+resource "helm_release" "this" {
+  count = 1 - local.argocd_enabled
+
+  depends_on = [
+    var.module_depends_on
+  ]
+
+  name          = var.release_name
+  repository    = var.chart_repository
+  chart         = var.chart_name
+  version       = var.chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+
+  dynamic "set" {
+    for_each = merge(local.conf)
+
+    content {
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+resource "local_file" "this" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content  = yamlencode(local.application)
+  filename = "${path.root}/${var.argocd.path}/${var.chart_name}.yaml"
+}
+
+
+locals {
+  conf       = merge(local.conf_defaults, var.conf)
+  conf_defaults = {
+    "namespace"                  = local.namespace
+    "rbac.create"                = true,
+    "resources.limits.cpu"       = "100m",
```

Comment:
are these limits sufficient for the monitoring server? Maybe they don't work or you should raise them a little

Reply:
I fixed resource limits.